src/wallet/rpcwallet.cpp

```diff
@@ -2540,15 +2554,57 @@ static UniValue loadwallet(const JSONRPCRequest& request)
     return obj;
 }
 
-static UniValue createwallet(const JSONRPCRequest& request)
+static UniValue setwalletflag(const JSONRPCRequest& request)
 {
+    std::shared_ptr<CWallet> const wallet = GetWalletForJSONRPCRequest(request);
+    CWallet* const pwallet = wallet.get();
+
+    if (!EnsureWalletIsAvailable(pwallet, request.fHelp)) {
+        return NullUniValue;
+    }
+
     if (request.fHelp || request.params.size() < 1 || request.params.size() > 2) {
+        std::string flags = "";
+        for (auto& it : walletflagmap) flags += (flags == "" ? "" : ", ") + it.first;
+        throw std::runtime_error(
+            "setwalletflag flag ( state )\n"
+            "\nChange the state of the given wallet flag for a wallet.\n"
+            "\nArguments:\n"
+            "1. flag    (string, required) The name of the flag to change. Current available flags: " + flags + "\n"
+            "2. value   (boolean, optional, default=true) The new state.\n"
+            "\nResult:\n"
+            "true|false (boolean) True or false, reflecting whether the flag was modified (true), or (false) whether it was already set to the given state\n"
+            "\nExamples:\n"
+            + HelpExampleCli("setwalletflag", "avoid_reuse")
+            + HelpExampleRpc("setwalletflag", "\"avoid_reuse\"")
+        );
+    }
+
+    std::string flag = request.params[0].get_str();
+    bool value = request.params[1].isNull() || request.params[1].get_bool();
+
+    if (!walletflagmap.count(flag)) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Unknown wallet flag %s", flag));
+    }
+
+    if (pwallet->IsWalletFlagSet(walletflagmap.at(flag))) return false;
+
+    pwallet->SetWalletFlag(walletflagmap.at(flag), value);
```

Comment:
I don't think this is sufficient for enabling 'avoid_reuse'. Consider:

- wallet is created without the `avoid_reuse` flag
- wallet sends from address
- wallet receives new coins to used address. These are NOT marked as dirty because the `avoid_reuse` flag is not set
- `avoid_reuse` is enabled with the `setwalletflag` RPC
- `getbalance`, `sendtoaddress`, etc will all use the reused address coins because they're not set as dirty.

Reply:
Right ... the idea is that the user has to actually call 'rescan' for it to become retroactive. This was explained when this was a command line option, I think, but the explanation vanished with the change to remove the option. Not sure where would be a good place to explain this. Perhaps there should be a 'caveats' map for flags which are returned when setting flags?